R/request_handler.R

```diff
@@ -24,21 +24,32 @@ RequestHandler <- R6::R6Class(
       if (private$externally_stubbed()) {
         # FIXME: not quite sure what externally_stubbed is meant for
         #   perhaps we can get rid of it here if only applicable in Ruby
-        vcr_log_sprintf("- externally stubbed")
-        private$on_externally_stubbed_request()
-      } else if (should_be_ignored(self$request)) {
-        vcr_log_sprintf("- ignored")
-        private$on_ignored_request()
-      } else if (cassette_has_response(self$request)) {
-        vcr_log_sprintf("- stubbed by vcr")
-        private$on_stubbed_by_vcr_request()
-      } else if (cassette_is_recording()) {
-        vcr_log_sprintf("- recordable")
-        private$on_recordable_request()
-      } else {
-        vcr_log_sprintf("- unhandled")
-        private$on_unhandled_request()
+        vcr_log_sprintf("  externally stubbed")
+        return(private$on_externally_stubbed_request())
       }
+
+      if (should_be_ignored(self$request)) {
+        vcr_log_sprintf("  ignored")
+        return(private$on_ignored_request())
+      }
+
+      if (cassette_active()) {
```

Comment:
I needed to refactor this so we only look for the request once (instead of once to see if it's available and then a second time to actually find it).

Reply:
looks good